build/azure-pipelines.release.yml

```diff
@@ -118,52 +120,81 @@ stages:
       - job: publish_binaries
         steps:
           - task: GoTool@0
+            condition: and(succeeded(), eq(variables.HAVE_SECRETS, 'true'))
```

Comment:
I tried a bunch of different ways to put the condition on the stage or job with zero success, so it's on every task...

Reply:
Hah, I just returned from looking up the Azure Pipelines docs to see if we could do!  Of course, you would've tried :)

Reply:
If you can figure it out, we can fix in a followup. I spent an entire day trying different variants (on a fake pipeline so it would fail fast) and was unable to get it to work when the condition relied transiently upon a secret.